File: Decorator/PHP/Whip.php
<?php
  class Whip extends CondimentDecorator {
    public $beverage;

    public function __construct($beverage) {
      $this->description = " Whip";
      $this->beverage = $beverage;
    }

    public function getDescription() {
      return $this->beverage->getDescription() . $this->description;
    }

    public function cost() {
      return $this->beverage->cost() + 9.9;
    }
  }

File: Decorator/Go/decorator/whip.go
package decorator

type Whip struct {
  *Condiment
}

func NewWhip(beverage beverage) *Whip {
  return &Whip{&Condiment{beverage: beverage, price: 0.9, description: "Whip"}}
}

func (self *Whip) Cost() float64 {
  return self.beverage.Cost() + self.price
}

func (self *Whip) GetDescription() string {
  return self.beverage.GetDescription() + " with " + self.description
}
